test/test_psutil.py

```diff
@@ -2475,6 +2475,60 @@ def test_Popen(self):
             proc.wait()
             self.assertIsNotNone(proc.returncode)
 
+    @unittest.skipUnless(hasattr(psutil.Process, "environ"),
+                         "environ not available")
+    def test_environ(self):
+        self.maxDiff = None
+        p = psutil.Process()
+        d = p.environ()
+        d2 = os.environ.copy()
+
+        if OSX:
+            for key in (
+                "__CF_USER_TEXT_ENCODING",
+                "VERSIONER_PYTHON_PREFER_32_BIT",
+                "VERSIONER_PYTHON_VERSION",
+            ):
+                d.pop(key, None)
+                d2.pop(key, None)
+
+        self.assertEqual(d, d2)
+
+    @unittest.skipUnless(hasattr(psutil.Process, "environ"),
+                         "environ not available")
+    @unittest.skipUnless(POSIX, "posix only")
+    def test_weird_environ(self):
+        # environment variables can contain values without an equals sign
+        # TODO: arbitrary binary data should also be tested but currently
+        # breaks on Python 3
+        code = textwrap.dedent("""
+        #include <unistd.h>
+        #include <fcntl.h>
+        char * const argv[] = {"cat", 0};
+        char * const envp[] = {"A=1", "X", "C=3", 0};
+        int main(void) {
+            /* Close stderr on exec so parent can wait for the execve to
+             * finish. */
+            if (fcntl(2, F_SETFD, FD_CLOEXEC) != 0)
+                return 0;
+            return execve("/bin/cat", argv, envp);
+        }
+        """)
+        path = create_temp_executable_file("x", code=code)
+        self.addCleanup(safe_remove, path)
+        sproc = get_test_subprocess([path],
+                                    stdin=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+        p = psutil.Process(sproc.pid)
+        wait_for_pid(p.pid)
+        self.assertTrue(p.is_running())
+        # Wait for process to exec or exit.
+        self.assertEquals(sproc.stderr.read(), "")
+        self.assertEquals(p.name(), "cat")
```

Comment:
I think this is not reliable across all POSIX platforms (name can be different) - I would remove it.